content/blog/2023-12-26-xmpp-providers-fully-automated.md

```diff
@@ -0,0 +1,75 @@
+
+**Additional new properties:**
+
+- `alternativeJids`: A list of JIDs a provider offers for registration other than the main JID.
+- `serverTesting`: Whether tests against the provider's server are allowed (e.g., certificate checks and uptime monitoring).
+
+## A Provider File For More Automation
+
+Not all of the properties collected by the XMPP Providers project are machine-readable yet.
+To enable automatic collection of the missing properties, the team works on extending existing standards and, if necessary, creating new ones.
+
+As a workaround until standards have been extended or created, and until those changes have been implemented and deployed in the wild, a 'provider file' shall fill the gap.
+A provider file is a JSON file containing only the provider properties that cannot be retrieved via other methods. Each provider can generate a provider file and supply it via its web server.
+To make this as easy as possible, a [Provider File Generator](/provider-file-generator/) has been developed, which generates a provider file from the information you supply.
+
+Example URL for a provider file of version 2:
+
+`https://example.org/.well-known/xmpp-provider-v2.json`
+
+As soon as a provider file is discovered by the tools, all properties listed in the provider file are automatically fetched and processed.
+
+#### Properties which can be fetched automatically via a provider file
+
+- Website: Provider website (per language).
+- Alternative JIDs: List of JIDs a provider offers for registration other than the main JID. New in API `v2`.
+- [Bus factor](https://en.wikipedia.org/wiki/Bus_factor): Bus factor of the XMPP service (i.e., the minimum number of team members that the service could not survive losing).
+- Organization: Type of organization providing the XMPP service. New in API `v2`, formerly `company`.
+- Password reset: Password reset web page (per language) used for an automatic password reset (e.g., via email) or describing how to manually reset a password (e.g., by contacting the provider).
+- Maximum HTTP File Upload total size: Maximum size of all shared files in total per user. This property is *not* about the maximum size of each shared file, which is already retrieved via XMPP.
+- Maximum HTTP File Upload storage time: Maximum storage duration of each shared file.
+- Maximum Message Archive Management storage time: Maximum storage duration of each exchanged message.
+- Professional hosting: Whether the XMPP server is hosted with good internet connection speed, uninterruptible power supply, access protection and regular backups.
+- Free of charge: Whether the XMPP service can be used for free.
+- Legal notice: Legal notice web page (per language).
+- Server locations: Server/Backup locations (countries).
+- Server testing: Whether tests against the provider's server are allowed (e.g., certificate checks and uptime monitoring). New in API `v2`.
+- Since: Date since the XMPP service is available.
```

Comment:
I suggest to only list the property changes.

Reply:
But these are not API changes, but properties provided by the provider file.